In [1]:
from matplotlib import pyplot as plt
import seaborn as sns
import math
import scipy
import numpy as np
import scipy.stats as st
import pandas as pd

# Узнаем номера моих задач

In [2]:
from zlib import crc32
import numpy as np

mail = input('\n')
to_hash =  mail.strip().split('@')[0]    
h = crc32(to_hash.encode('utf-8'))
seed = h%(2**32-1)
rs = np.random.RandomState(seed)
task1 = rs.randint(low=1, high=5)
task2 = rs.randint(low=1, high=5)
task3 = rs.randint(low=1, high=5)
task4 = rs.randint(low=1, high=4)
print ('Your tasks are 1.{}, 2.{}, 3.{}, 4.{}'.format(task1, task2, task3, task4))


ugnivenko.va@phystech.edu
Your tasks are 1.1, 2.2, 3.2, 4.3


# Задача 4.3 

Правительство города М. испытывает систему обнаружения нежелаемых лиц по камерам в метро. В качестве демонстрации работоспособности системы была поставлена цель: найти и задержать опасного рецидивиста по имени Николай Вальный, а также его соучастников. Была собрана выборка из 5000 снимков лица, для которых была проверена гипотеза о несовпадении этого снимка с лицами участников команды Н. Вального. Для 100 фотографий нулевая гипотеза была отвергнута на уровне значимости alpha = 0.05.

Требуется:

- Определить, в чем недостаток данного подхода и как можно его улучшить.
- Предложить наилучший, на ваш взгляд, способ для повышения качества данного решения.
- Какую меру качества контролирует данный способ? Какие гарантии он предоставляет?
- В чём недостатки данного способа?
- Как изменилась мощность при использовании предложенного вами способа относительно изначальной процедуры проверки?
- Известно, что все 5000 фотографий были сделаны для разных людей, и правительство хочет, чтобы система ни в коем случае не упустила преступников. Ответьте на те же вопросы из пунктов 2-4.
- Как изменилась мощность при использовании предложенного вами способа относительно предыдущего способа?

# Ответы 

- Недостаток этого способа состоит в том, что не учли поправку на множественную проверку гипотез. Чтобы улучшить данный способ, надо ввести эту поправку.
- 100 фотографий - довольно много, поэтому стоит ввести самую мощную поправку - Holm. 
- Он контролирует метрику качества FWER. FWER - это групповая вероятность ошибки первого рода. Метод Holm  гарантирует, что вероятность совершения хотя бы одной ошибки первого рода не превосходит $\alpha$.
- Данный метод имеет низкую мощность - это следствие контроля FWER на уровне значимости $\alpha$.
- Мощность станет ниже.
- Тогда в данном случае, лучше рассматривать метрику качества FDR - ожидаемая доля ложных отклонений гипотез. Так как теперь мы знаем, что статистики - независимы, то можно использовать поправку Бенджамини-Хохберга.
- Мощность увеличилась, так как смягчилось условия на количество ошибок первого рода.